core-java-modules/core-java-networking-2/src/main/java/com/baeldung/url/auth/HttpClient.java

```diff
@@ -0,0 +1,47 @@
+package com.baeldung.url.auth;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.ProtocolException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.commons.codec.binary.Base64;
+
+public class HttpClient {
+
+    private String baseUrl;
+    private String user;
+    private String password;
+
+    public HttpClient(String baseUrl, String user, String password) {
+        this.baseUrl = baseUrl;
+        this.user = user;
+        this.password = password;
+    }
+
+    public int sendRquestWithBasicAuth(String path) throws IOException {
+        HttpURLConnection connection = null;
+        try {
+            connection = createConnection(path);
```

Comment:
let's move this to a separate method to improve readability, something like String createBasicAuthHeaderValue(String u, String p) then you do something like 
connection.setRequestProperty("Authorization", createBasicAuthHeaderValue("u", "p"));